DQM/EcalMonitorTasks/src/PiZeroTask.cc

```diff
@@ -0,0 +1,287 @@
+#include "DQM/EcalMonitorTasks/interface/PiZeroTask.h"
+
+namespace ecaldqm {
+  PiZeroTask::PiZeroTask()
+      : DQWorkerTask(),
+        seleXtalMinEnergy_(0.f),
+        clusSeedThr_(0.f),
+        clusEtaSize_(0),
+        clusPhiSize_(0),
+        selePtGammaOne_(0.f),
+        selePtGammaTwo_(0.f),
+        seleS4S9GammaOne_(0.f),
+        seleS4S9GammaTwo_(0.f),
+        selePtPi0_(0.f),
+        selePi0Iso_(0.f),
+        selePi0BeltDR_(0.f),
+        selePi0BeltDeta_(0.f),
+        seleMinvMaxPi0_(0.f),
+        seleMinvMinPi0_(0.f),
+        posCalcParameters_(edm::ParameterSet()) {}
+
+  void PiZeroTask::setParams(edm::ParameterSet const& params) {
```

Comment:
Were is this `setParams` method called? Should it be in the constructor? 

Can you also define a fillDescription method, to initialize and validated all those parameters?

Reply:
> Were is this `setParams` method called? Should it be in the constructor?

The inheritance structure of ECAL DQM can be non-trivial at times. The ECAL DQM analyzer being run is `EcalDQMonitorTask`, which itself executes `DQWorker`s such as `PiZeroTask`. The analyzer parent class `EcalDQMonitor` constructor passes the parameters to the worker here:
https://github.com/cms-sw/cmssw/blob/c488deecab052ff8f5e4eeed6e122b55589de505/DQM/EcalCommon/src/EcalDQMonitor.cc#L35-L36
which are then set here:
https://github.com/cms-sw/cmssw/blob/c488deecab052ff8f5e4eeed6e122b55589de505/DQM/EcalCommon/src/DQWorker.cc#L194

Reply:
> Can you also define a fillDescription method, to initialize and validate all those parameters?

`fillDescriptions()` is already called but is again non-trivial to see. The ECAL DQM analyzer has it's own `fillDescriptions()` which calls
https://github.com/cms-sw/cmssw/blob/c488deecab052ff8f5e4eeed6e122b55589de505/DQM/EcalMonitorTasks/plugins/EcalDQMonitorTask.cc#L85
which calls
https://github.com/cms-sw/cmssw/blob/c488deecab052ff8f5e4eeed6e122b55589de505/DQM/EcalMonitorTasks/src/DQWorkerTask.cc#L11
which finally calls
https://github.com/cms-sw/cmssw/blob/c488deecab052ff8f5e4eeed6e122b55589de505/DQM/EcalCommon/src/DQWorker.cc#L50